beacon-chain/forkchoice/doubly-linked-tree/types.go

```diff
@@ -38,29 +40,38 @@ type Store struct {
-	children                 []*Node                      // the list of direct children of this Node
-	justifiedEpoch           primitives.Epoch             // justifiedEpoch of this node.
-	unrealizedJustifiedEpoch primitives.Epoch             // the epoch that would be justified if the block would be advanced to the next epoch.
-	finalizedEpoch           primitives.Epoch             // finalizedEpoch of this node.
-	unrealizedFinalizedEpoch primitives.Epoch             // the epoch that would be finalized if the block would be advanced to the next epoch.
-	balance                  uint64                       // the balance that voted for this node directly
-	weight                   uint64                       // weight of this node: the total balance including children
-	bestDescendant           *Node                        // bestDescendant node of this node.
-	optimistic               bool                         // whether the block has been fully validated or not
-	timestamp                uint64                       // The timestamp when the node was inserted.
+	slot                         primitives.Slot                      // slot of the block converted to the node.
+	root                         [fieldparams.RootLength]byte         // root of the block converted to the node.
+	payloadHash                  [fieldparams.RootLength]byte         // payloadHash of the block converted to the node.
+	parent                       *Node                                // parent index of this node.
+	target                       *Node                                // target checkpoint for
+	children                     []*Node                              // the list of direct children of this Node
+	justifiedEpoch               primitives.Epoch                     // justifiedEpoch of this node.
+	unrealizedJustifiedEpoch     primitives.Epoch                     // the epoch that would be justified if the block would be advanced to the next epoch.
+	finalizedEpoch               primitives.Epoch                     // finalizedEpoch of this node.
+	unrealizedFinalizedEpoch     primitives.Epoch                     // the epoch that would be finalized if the block would be advanced to the next epoch.
+	balance                      uint64                               // the balance that voted for this node directly
+	weight                       uint64                               // weight of this node: the total balance including children
+	bestDescendant               *Node                                // bestDescendant node of this node.
+	optimistic                   bool                                 // whether the block has been fully validated or not
+	timestamp                    uint64                               // The timestamp when the node was inserted.
+	latestExecutionPayloadHeader *enginev1.ExecutionPayloadHeaderEPBS // latestExecutionPayloadHeader stores the latest execution payload header for this node
```

Comment:
We can't really keep these structures in forkchoice, the spec version is highly not optimized